packages/charts/src/components/chart/chart.tsx

```diff
@@ -0,0 +1,417 @@
+import {
+  useRef,
+  useMemo,
+  forwardRef,
+  useCallback,
+  type ComponentPropsWithRef,
+  useEffect,
+} from 'react'
+import type { EChartsOption, SeriesOption } from 'echarts'
+import ReactECharts from 'echarts-for-react'
+import * as echarts from 'echarts'
+import './locales'
+import { defaultTheme } from '../../theme/themes'
+import type { ChartConfig, DefaultHooks } from '../../types/chart'
+import {
+  checkValidVariant,
+  checkZoom,
+  getChartOptions,
+  getDefaultByType,
+} from '../../utils/chart'
+import { canUseDOM, useMergeRef } from '@vtex/shoreline-utils'
+import {
+  DATAZOOM_DEFAULT_STYLE,
+  LEGEND_DEFAULT_STYLE,
+} from '../../theme/chartStyles'
+import { cloneDeep, isArray, type Dictionary } from 'lodash'
+import {
+  formatTimeAxis,
+  normalizeBarData,
+  normalizeHorizontalBarData,
+  normalizeStackedBars,
+  roundCap,
+  setAreaColors,
+  setAreaGradients,
+  sunburstCoreColoring,
+} from '../../utils/hooks'
+import {
+  toggleSerieLegend,
+  turnOffSerieLegend,
+  turnOnAllLegend,
+  turnOnSerieLegend,
+} from '../../utils/legend'
+import {
+  Legend,
+  type LegendHandle,
+  type LegendAction,
+  getChanges,
+  handleChanges,
+} from '../legend'
+
+import '../../theme/components/chart.css'
+import { ChartSkeleton } from '../chart-skeleton/chart-skeleton'
+
+/**
+ * Render a Shoreline Chart with Echarts. Mixes user options with defaults determined by chart type.
+ * @see https://echarts.apache.org/en/index.html
+ * @example
+ * <Chart
+      chartConfig={{ type: 'bar', variant: 'default' }}
+      option={{
+        xAxis: {
+          data: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'],
+        },
+        series: [{ data: [1, 2, 3, 4, 5, 6, 7] }],
+      }}
+      style={{ height: 550 }}
+    />
+ */
+export const Chart = forwardRef<ReactECharts | undefined, ChartProps>(
+  function Charts(props, ref) {
+    const {
+      series,
+      xAxis = {},
+      yAxis = {},
+      title,
+      option,
+      loading = false,
+      loadingOptions,
+      chartConfig,
+      style,
+      renderer = 'svg',
+      locale = 'en',
+      theme = defaultTheme,
+      optionHooks = [],
+      onEvents,
+      zoom,
+      checkboxLegend = true,
+      group,
+      ...otherProps
+    } = props
+
+    const chartRef = useRef<ReactECharts>(null)
+    const legendRef = useRef<LegendHandle>(null)
+
+    const hooks: ((series: EChartsOption) => EChartsOption)[] = useMemo(() => {
+      if (optionHooks === null || chartConfig === null) {
+        return []
+      }
+      const { type, variant } = chartConfig
+      const checkedVariant =
+        variant && checkValidVariant(type, variant)
+          ? variant
+          : getDefaultByType(type)
+
+      const hooks: any[] = []
+      if (!isArray(xAxis) && xAxis.type === 'time') {
+        hooks.push(formatTimeAxis(locale))
+      }
+      hooks.push(...defaultHooks[type][checkedVariant])
+      hooks.push(...optionHooks)
+      return hooks
```

Comment:
Take care with the readability. Add break lines between scopes like, between ifs and returns.